lib/cuckoo/common/web_utils.py

```diff
@@ -1249,16 +1249,15 @@ def process_new_task_files(request, samples, details, opt_filename, unique):
         size = sample.size
         data = False
         if size > web_cfg.general.max_sample_size:
-            if not (web_cfg.general.allow_ignore_size and "bypass_size_check" in details["options"]):
+            if not (web_cfg.general.allow_ignore_size and "ignore_size_check" in details["options"]):
                 first_chunk = sample.chunks().__next__()
                 if web_cfg.general.enable_trim and HAVE_PEFILE and IsPEImage(first_chunk):
                     trimmed_size = trim_sample(sample.chunks().__next__())
                     if trimmed_size:
                         size = trimmed_size
-                        data = sample.chunks(size).__next__()
-                else:
-                    # we need to rebuild original file
-                    data = first_chunk + sample.read()
+                # move to start of sample file
+                sample.seek(0)
+                data = sample.read(size)
```

Comment:
I think these 2 lines should go after the following 'if' block so that we don't read a huge file in to memory if we're just going to skip it anyway.

Reply:
Agreed.